bundle/jsky.app.ot/src/main/scala/jsky/app/ot/gemini/parallacticangle/ParallacticAngleControls.scala

```diff
@@ -161,13 +165,34 @@ class ParallacticAngleControls(isPaUi: Boolean) extends GridBagPanel with Publis
     } {
       val spObs = ispObs.getDataObject.asInstanceOf[SPObservation]
       val sameNight = spObs.getSchedulingBlock.asScalaOpt.exists(_.sameObservingNightAs(sb))
-      spObs.setSchedulingBlock(ImOption.apply(sb))
-      ispObs.setDataObject(spObs)
-      if (!sameNight) {
-        EphemerisUpdater.unsafeRefreshEphemerides(ispObs, e.getWindow)
-      }
-      callback.run()
-      resetComponents()
+
+      // IO action on EDT
+      def edt[A](a: => Unit): IO[Unit] =
+        IO(Swing.onEDT(a))
+
+      // The update we want to do
+      val action: IO[Unit] =
+        for {
+          _ <- IO(spObs.setSchedulingBlock(ImOption.apply(sb)))
+          _ <- IO(ispObs.setDataObject(spObs))
+          _ <- sameNight.unlessM(EphemerisUpdater.refreshEphemerides(ispObs, e.getWindow))
```

Comment:
The UI is blocked the whole time so I'm not sure it matters that much, but I guess I could batch up all the edits and do them at the end?